File: 293-Flip-Game/solution.java
public class Solution {
    public List<String> generatePossibleNextMoves(String s) {
        List<String> result = new ArrayList<String>();
        if(s == null || s.length() <= 1) return result;
        if(s.length() == 2 && !s.equals("++")) return result;
        int n = s.length();
        for(int i = 1; i < n; i++) {
            char pre = s.charAt(i - 1);
            char cur = s.charAt(i);
            if(pre == cur && pre == '+') {
                result.add(s.substring(0, i -1) + "--" + s.substring(i + 1, n));
            }
        }
        return result;
    }
}

File: 293-Flip-Game/solution.py
class Solution(object):
    def generatePossibleNextMoves(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        result = []
        for i in range(1, len(s)):
            if s[i] == '+' and s[i - 1] == '+':
                result.append(s[:i-1] + "--" + s[i+1:])
        return result